### PR TITLE
feat: add sendBufferSize option

### PR DESCRIPTION
**Pull request by maxence2997**
## Summary

Add `sendBufferSize` option to make the outbound channel capacity configurable. Previously hardcoded at 256, now matches server and client-go's `WithSendBufferSize` API.

## Changes

- Add `sendBufferSize` to `ClientConfig` DSL (default 256, range [1, 4096])
- Add validation in `validateConfig`: throws `IllegalArgumentException` if < 1 or > 4096
- Replace hardcoded channel capacity with configured value
- Add tests: valid values, zero/negative/exceeds-max throws

### Required

- [x] `make check` passes (lint → test)
- [x] Each commit represents exactly one logical change
- [x] Commit messages follow the format in `commit-message-instructions.md`
- [x] No unrelated code reformatting in this PR

### Conditional

- [x] New public API: includes KDoc comments

### CODE REVIEW

File: src/test/kotlin/com/wspulse/client/ConfigValidationTest.kt [hunk 1]
```diff
@@ -14,6 +14,67 @@ import kotlin.time.Duration.Companion.seconds
...
+    fun `sendBufferSize 1 is valid`() {
+        val ex =
+            assertThrows<Exception> {
+                runBlocking {
+                    WspulseClient.connect("ws://127.0.0.1:1") {
+                        sendBufferSize = 1
+                    }
+                }
+            }
+        assertFalse(ex is IllegalArgumentException, "sendBufferSize=1 should be valid")
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/test/kotlin/com/wspulse/client/ConfigValidationTest.kt:29]: Won't fix. Kotlin `require()` throws `IllegalArgumentException` at the start of `connect()` before any network call. The test correctly asserts that valid values do NOT throw `IllegalArgumentException` — if validation passed, any subsequent exception is a connection failure (expected without a server). The test is deterministic: `require()` is synchronous and runs before coroutine launch.

File: src/main/kotlin/com/wspulse/client/WspulseClient.kt [hunk 2]
```diff
@@ -589,6 +587,13 @@ class WspulseClient private constructor(
  * are allocated so invalid config never leaks an [HttpClient].
  */
 private fun validateConfig(config: ClientConfig) {
+    require(config.sendBufferSize >= 1) {
+        "wspulse: sendBufferSize must be at least 1"
+    }
+    require(config.sendBufferSize <= MAX_SEND_BUFFER_SIZE) {
+        "wspulse: sendBufferSize exceeds maximum (4096)"
```
[the remark this answers is not in the document]
Comment (maxence2997) [reply src/main/kotlin/com/wspulse/client/WspulseClient.kt]: Fixed in b0d6c86. Error message now uses \$MAX_SEND_BUFFER_SIZE string interpolation.
